File: project/cheolhyun-onboarding/survey-domain/src/test/java/survey/domain/model/OptionTest.java
package survey.domain.model;

import com.onboarding.model.Option;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.*;
import static org.junit.jupiter.api.Assertions.*;

class OptionTest {
    @Test
    @DisplayName("null과 빈값 유효성 검증")
    void 옵션_빈값_검증() {
        assertAll(
                () -> assertThatThrownBy(() -> new Option("")).isInstanceOf(IllegalArgumentException.class),
                () -> assertThatThrownBy(() -> new Option(null)).isInstanceOf(IllegalArgumentException.class)
        );
    }
}

File: project/jihochoi-onboarding/QueryBox/src/test/kotlin/com/wlghsp/querybox/domain/survey/OptionTest.kt
package com.wlghsp.querybox.domain.survey

import org.assertj.core.api.Assertions.assertThatThrownBy
import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class OptionTest {

    @DisplayName("옵션 값이 정상적으로 설정되면 객체가 생성 성공")
    @Test
    fun optionCreateTest() {
        val value = "서울"

        val option = Option(value)

        assertThat(option.text).isEqualTo(value)
    }

    @DisplayName("옵션 값이 비어 있으면 예외 발생")
    @Test
    fun optionCreateExceptionTest() {
        assertThatThrownBy { Option("") }
            .isInstanceOf(IllegalArgumentException::class.java)
            .hasMessageContaining("옵션 값은 비어 있을 수 없습니다.")
    }
}
